### PR TITLE
Pass Retirement from App to Site and generate dynamic Pages

### PR DESCRIPTION
**Pull request by ladytrekker**
## Description

This PR does:
- App: get the total retirement number of an address
- App: set the total retirement number to the offset state
- App: Show external Link to Site with the beneficiary address and total retirement number **(Staging only)**
- Site: Generate dynamic pages from URL `retirement/<ADDRESS>` **(Staging only)**
- Site: Generate dynamic pages from URL `retirement/<ADDRESS>/<INDEX>` **(Staging only)**

The implemented contract methods on Site are **placeholders** until we get the actual GraphQL queries working from subgraph. **See the list of further steps below.**

## HOW TO TEST

1. Do an Offset here:
https://klimadao-app-git-retirement-pages-klimadao.vercel.app/#/offset
2. Wait for the Success Modal
3. In the Success Modal at the bottom copy the URL from the link "Open retirement details on: LINK`
4. In the copied URL replace `https://klimadao.finance/<ADDRESS>/<INDEX>` with the URL from this PR preview link for SITE: `https://klimadao-site-git-retirement-pages-klimadao.vercel.app/<ADDRESS>/<INDEX>`
5. Paste this new URL into your browser and let NextJS generate the result for you

## Further steps
 
- [x] Remove the contract methods used in Site for generating the pages **=>** [see PR](https://github.com/KlimaDAO/klimadao/pull/367)
- [x] Instead query the subgraph with ADDRESS and INDEX from [this query](https://thegraph.com/hosted-service/subgraph/klimadao/polygon-bridged-carbon?query=Recent%20KlimaDAO%20Retirements) **=>**  [see PR](https://github.com/KlimaDAO/klimadao/pull/367)
- [x] Pass the data to the dynamic pages [ **=>**  see PR](https://github.com/KlimaDAO/klimadao/pull/367)
- [ ] Style and build the pages according to Figma Designs

## Related Ticket

Part of https://github.com/KlimaDAO/klimadao/issues/345

## Checklist

- [x] Building the site with `npm run build-site` works without errors
- [x] Building the app with `npm run build-app` works without errors
- [x] I formatted JS and TS files with running `npm run format-all`

### CODE REVIEW

File: site/pages/retirement/[beneficiary_address]/[retirement_index].tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+import { getRetirementIndexInfo } from "@klimadao/lib/utils";
+import { SingleRetirementPage } from "components/pages/Retirement/SingleRetirement";
+import { loadTranslation } from "lib/i18n";
+import { IS_PRODUCTION } from "lib/constants";
+
+export const getStaticProps: GetStaticProps = async (ctx) => {
+  try {
+    if (IS_PRODUCTION) {
+      throw new Error("Not on Staging");
```
Comment (Atmosfearful) [inline site/pages/retirement/[beneficiary_address]/[retirement_index].tsx:11]: return `props.notFound` is also an option

File: site/pages/retirement/[beneficiary_address]/index.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  try {
+    if (IS_PRODUCTION) {
+      throw new Error("Not on Staging");
+    }
+
+    const { params, locale } = ctx;
+
+    if (!params || !params?.beneficiary_address) {
+      throw new Error("No params found");
+    }
```
Comment (Atmosfearful) [inline site/pages/retirement/[beneficiary_address]/index.tsx:19]: as I later cleanup you can pass a type argument to GetStaticProps for the expected params

**Review by Atmosfearful — APPROVED** [review]
beauuuutiful

**Comment by ShimonD-KlimaDAO** [comment]
This doesn't seem to be working. @ladytrekker I DM'd you my link in the form:
https://klimadao-site-git-retirement-pages-klimadao.vercel.app/<ADDRESS>/<INDEX>

File: app/components/views/Offset/index.tsx [hunk 3]
```diff
@@ -677,6 +684,17 @@ const RetirementSuccessModal = (props: RetirementSuccessModalProps) => {
               </A>
             </Trans>
           </Text>
+          {!IS_PRODUCTION && props.retirementTotals && (
+            <Text t="caption">
+              Open retirement details on:
+              <A
+                target="_blank"
+                href={`${urls.home}/retirement/${props.beneficiaryAddress}/${props.retirementTotals}`}
```
[the remark this answers is not in the document]
Comment (ladytrekker) [reply app/components/views/Offset/index.tsx:692]: Hm, it should point to Site and not App.
`urls.home` is the Site URL.
Yes? No?

**Comment by ladytrekker** [comment]
> This doesn't seem to be working. @ladytrekker I DM'd you my link in the form: https://klimadao-site-git-retirement-pages-klimadao.vercel.app/ADDRESS/INDEX

Thanks for checking this PR

Some retirements throw an error when fetching the data.

You can find the fix in this PR => https://github.com/KlimaDAO/klimadao/pull/369

File: site/pages/retirement/[beneficiary_address]/[retirement_index].tsx (hunk 1 above)
Comment (ladytrekker) [reply site/pages/retirement/[beneficiary_address]/[retirement_index].tsx:11]: I thought to keep the string as we can later see on the Vercel logs the reason why the page was not generated / resulted in 404.

File: site/pages/retirement/[beneficiary_address]/index.tsx (hunk 2 above)
Comment (ladytrekker) [reply site/pages/retirement/[beneficiary_address]/index.tsx:19]: Will do in follow-up 👍

**Pull request merged**